src/algorithms/ctmrg.jl

```diff
@@ -145,12 +140,33 @@ function gauge_fix(envprev::CTMRGEnv{C,T}, envfinal::CTMRGEnv{C,T}) where {C,T}
             scalartype(T),
             MPSKit._lastspace(Tsfinal[end])' ← MPSKit._lastspace(M[end])',
         )
-        ρprev = eigsolve(TransferMatrix(Tsprev, M), ρinit, 1, :LM)[2][1]
-        ρfinal = eigsolve(TransferMatrix(Tsfinal, M), ρinit, 1, :LM)[2][1]
+
+        # this is a bit of a hack to get the fixed point of the mixed transfer matrix
+        # because MPSKit is not compatible with AD
+        ρ_prev = let Tsprev = Tsprev, M = M
```

Comment:
Probably better to split the contents of the `let` block off into a separate function and then call it twice instead of having basically the same block twice?